### PR TITLE
reference: refine DM TOC

### PR DESCRIPTION
**Pull request by ericsyh**
### What is changed, added or deleted? 

There are some useful DM documents not included in TOC which is not easy for users to touch them.

### CODE REVIEW

File: dev/TOC.md [hunk 1]
```diff
@@ -98,14 +100,20 @@
       + 配置
         - [概述](/dev/reference/tools/data-migration/configure/overview.md)
         - [任务配置](/dev/reference/tools/data-migration/configure/task-configuration-file.md)
+      + [DM 集群管理]
```
Comment (csuzhangxc) [inline dev/TOC.md]:
```suggestion
      + DM 集群管理
```

File: dev/TOC.md [hunk 2]
```diff
@@ -98,14 +100,20 @@
...
+        - [DM 集群操作](/dev/reference/tools/data-migration/cluster-operations.md)
+        - [DM 集群升级](/dev/reference/tools/data-migration/dm-upgrade.md)
+      + [DM 同步任务管理]
```
Comment (csuzhangxc) [inline dev/TOC.md]:
```suggestion
      + DM 同步任务管理
```

File: dev/TOC.md [hunk 3]
```diff
@@ -98,14 +100,20 @@
...
+        - [任务状态查询](/dev/reference/tools/data-migration/query-status.md)
       - [监控 DM 集群](/dev/reference/tools/data-migration/monitor.md)
-      - [管理数据同步任务](/dev/reference/tools/data-migration/manage-tasks.md)
-      - [DM 集群操作](/dev/reference/tools/data-migration/cluster-operations.md)
       + 从与 MySQL 兼容的数据库迁移数据
         - [从 Amazon Aurora MySQL 迁移数据](/dev/how-to/migrate/from-aurora.md)
       - [DM Portal](/dev/reference/tools/data-migration/dm-portal.md)
-      - [DM 错误含义和诊断](/dev/reference/tools/data-migration/error-system.md)
-      - [DM 故障诊断](/dev/how-to/troubleshoot/data-migration.md)
+      + [故障诊断]
```
Comment (csuzhangxc) [inline dev/TOC.md]:
```suggestion
      + 故障诊断
```

**Comment by lilin90** [comment]
@ericsyh Please address comments.

**Comment by ericsyh** [comment]
@lilin90 Updated, PTAL

**Comment by csuzhangxc** [comment]
@ericsyh update v2.1 and v3.0 too?

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**